NO-SNOW Docs housecleaning (#1356)
snowflakedb/snowflake-connector-net: doc/Client/QueryingData.md

**Note**: There are 2 levels of asynchronous execution. One is asynchronous execution in terms of C# language (`async await`).
Another is asynchronous execution of the query by the server (you can recognize it by `InAsyncMode` containing method names, e. g. `ExecuteInAsyncMode`, `ExecuteAsyncInAsyncMode`).

Example of asynchronous code starting a query to be executed asynchronously on the server:
```cs
using var conn = new SnowflakeDbConnection("account=testaccount;username=testusername;password=testpassword")
await conn.OpenAsync(CancellationToken.None);
SnowflakeDbCommand cmd = (SnowflakeDbCommand)conn.CreateCommand()
cmd.CommandText = "SELECT ...";
var queryId = await cmd.ExecuteAsyncInAsyncMode(CancellationToken.None);
      // ...
```

> You can check the status of a query executed asynchronously on the server:

Reviewer: why examples with blocking calls were removed?

Author: They don't present anything different than the async ones. It's standard to have async and non-async counterpart that do the same thing, adding sync examples only here, whilst other .md mention only the default and recommended async approach could be confusing. And adding it everywhere and effectively doubling the scope of the docs provided little value, as there's little informative value in those examples, since they are exactly the same in terms of usage, apart from the Async part.

The following example sets the MULTI_STATEMENT_COUNT session parameter to 1. Then for an individual command, it sets MULTI_STATEMENT_COUNT=3 to indicate that the query contains precisely three SQL commands. The query string, `cmd.CommandText` , then contains the three statements to execute.

```cs
using var conn = new SnowflakeDbConnection()
conn.ConnectionString = ConnectionString;
await conn.OpenAsync(cancellationToken);
var cmd = conn.CreateCommand();
cmd.CommandText = "ALTER SESSION SET MULTI_STATEMENT_COUNT = 1";
await cmd.ExecuteNonQueryAsync(cancellationToken);

> using var cmd2 = conn.CreateCommand()

Reviewer 2: ```suggestion
using var cmd2 = conn.CreateCommand();
```

// Set statement count

> var stmtCountParam = cmd.CreateParameter();

Reviewer 2: `cmd` =-> `cmd2`?

await conn.OpenAsync(cancellationToken);

var cmd = conn.CreateCommand();
cmd.CommandText = "create or replace table T(cola int)";
var count = cmd.ExecuteNonQuery();
Assert.AreEqual(0, count);

var cmd = conn.CreateCommand();
cmd.CommandText = "insert into t values (?), (?), (?)";

var p1 = cmd.CreateParameter();
p1.ParameterName = "1";
p1.Value = 10;
p1.DbType = DbType.Int32;
cmd.Parameters.Add(p1);

var p2 = cmd.CreateParameter();
p2.ParameterName = "2";
p2.Value = 10000L;
p2.DbType = DbType.Int32;
cmd.Parameters.Add(p2);

var p3 = cmd.CreateParameter();
p3.ParameterName = "3";
p3.Value = (short)1;
p3.DbType = DbType.Int16;
cmd.Parameters.Add(p3);

var count = cmd.ExecuteNonQuery();
Assert.AreEqual(3, count);

cmd.CommandText = "drop table if exists T";
count = cmd.ExecuteNonQuery();
Assert.AreEqual(0, count);

await conn.CloseAsync(cancellationToken);

```

## Bind Array Variables

The sample code creates a table with a single integer column and then uses array binding to populate the table with values 0 to 70000.

```cs
using var conn = new SnowflakeDbConnection()
conn.ConnectionString = ConnectionString;
await conn.OpenAsync(cancellationToken);

> using var cmd = conn.CreateCommand()

Reviewer 2: ```suggestion
using var cmd = conn.CreateCommand();
```

> cmd.CommandText = "create or replace table putArrayBin(colAinteger)";

Reviewer 2: ```suggestion
cmd.CommandText = "create or replace table putArrayBind(colAinteger)";
```

await cmd.ExecuteNonQueryAsync(cancellationToken);

string insertCommand = "insert into putArrayBind values (?)";
cmd.CommandText = insertCommand;

int total = 70000;

List<int> arrint = new List<int>();
for (int i = 0; i < total; i++)
{
	arrint.Add(i);
}
var p1 = cmd.CreateParameter();
p1.ParameterName = "1";
p1.DbType = DbType.Int16;
p1.Value = arrint.ToArray();
cmd.Parameters.Add(p1);

> count = await cmd.ExecuteNonQuery(cancellationToken); // count = 70000

Reviewer 2: ```suggestion
count = await cmd.ExecuteNonQueryAsync(cancellationToken); // count = 70000
```

using Newtonsoft.Json;
..

                    using var cmd = conn.CreateCommand()
                    var vals = new int[] { 1, 2, 3 };

>                     var array = JsonConvert.SerializeObjec(vals); // alternatively you can do `vals.ToArra()` when passing it to `p1.Value`

Reviewer 2: ```suggestion
                    var array = JsonConvert.SerializeObject(vals); // alternatively you can do `vals.ToArray()` when passing it to `p1.Value`
```

Author: yeah, I missed a changeset in stage, fixing those and didn't commit it, thanks for pointing it out!

                    var sql = "CALL test_db.public.test(parse_json(?))"; // test SP, returns a single value
                    // execute this sql with bind variable 'array'
                    cmd.CommandText = sql;

                    var p1 = cmd.CreateParameter();
                    p1.ParameterName = "1";
                    p1.Value = array; // passing the array in thebind variable. 
                    p1.DbType = DbType.String;
                    cmd.Parameters.Add(p1);

                    await using reader = cmd.ExecuteReaderAsync(cancellationToken);
                    while (await reader.ReadAsync(cancellationToken))
                    {
                        Console.WriteLine(reader.GetString(0));
                    }
                    await conn.CloseAsync(cancellationToken);
```

Writing to `VARIANT` column with bind variable
----------------------------------------------
You can use bind variables together with `VARIANT` datatype column. For example, to insert data:
```cs
var cmd = conn.CreateCommand();

> cmd.CommandText = "insert into table (id, data) select parse_json((?))"

Reviewer 2: `table` is reserved, if someone just copy/paste/run will get syntax error

```suggestion
cmd.CommandText = "insert into my_table (id, data) select parse_json((?))"
```

Author: great observation, fix'd
